pkg/dockerutil/dockerutils.go

```diff
@@ -839,6 +849,14 @@ func CheckDockerVersion(versionConstraint string) error {
 		}
 		return fmt.Errorf("%s", msgs)
 	}
+
+	dockerAPIVersion, err := GetDockerAPIVersion()
+	if err != nil {
+		return fmt.Errorf("unable to determine Docker API version: %v", err)
+	}
+	if !dockerVersions.GreaterThanOrEqualTo(dockerAPIVersion, minimalDockerAPIVersion) {
+		return fmt.Errorf("your Docker API version (%s) doesn't meet the minimum (%s) - see %s for version matrix (%s) for more help", dockerAPIVersion, minimalDockerAPIVersion, "https://docs.docker.com/reference/api/engine/#api-version-matrix", "https://ddev.rtfd.io/page/users/install/docker-installation/#testing-and-troubleshooting-your-docker-installation")
```

Comment:
Right now we have an optional check for `"1.44"` in our code, to modify template for the app:

https://github.com/ddev/ddev/blob/30fb117cb664273b085477edaacb76ed25166c5a/pkg/ddevapp/app_compose_template.yaml#L85-L87

https://github.com/ddev/ddev/blob/30fb117cb664273b085477edaacb76ed25166c5a/pkg/ddevapp/utils.go#L198-L211

If we change
```diff
-const minimalDockerAPIVersion = "1.43"
+const minimalDockerAPIVersion = "1.44"
```
and remove `templateCanUse`, making `start_interval: "1s"` available all the time, it will break `ddev start` for older Docker, because it'll simply show a warning, but it should fail completely.

---

So it should call `util.Failed` instead of returning an error.

Reply:
But didn't we decide that we can't go to 1.44 because we don't want to break people?

Reply:
Yes, but I'm thinking ahead 😅

When we decide to increase the `minimalDockerAPIVersion`, it'll simply show a warning for people with lower API version, it won't fail, which mean `ddev start` will continue to run, failing in another place.

Is it enough to show a warning in this case?

Reply:
You can change the `Version` shown in the `docker version` output to anything you want by specifying it as a build argument (which is why I believe we show it as a warning). Different Linux distributions can bundle Docker using various techniques, making this value unreliable.

However, the `API version` in `docker version` cannot be altered through build arguments, so it's something we can rely on. That's why I suggest we implement a hard failure if the API version doesn't meet the requirements.